lib/Dialect/TritonGPU/Transforms/AccelerateMatmul.cpp

```diff
@@ -452,27 +454,38 @@ class ScaledBlockedToMMAv2
     auto newAcc =
         rewriter.create<ConvertLayoutOp>(oldAcc.getLoc(), newRetType, oldAcc);
 
-    auto toMMABf16 = [&newRetType, &rewriter, &ctx,
-                      &enumToType](TypedValue<RankedTensorType> v, int idx,
-                                   F8F6F4Type type) {
-      // MMAv2 Layout
+    auto toMMABf16 = [&newRetType, &rewriter, &ctx, &enumToType](
+                         TypedValue<RankedTensorType> v, int idx,
+                         F8F6F4Type type) -> TypedValue<RankedTensorType> {
       auto vType = v.getType();
-      auto newVEncoding = DotOperandEncodingAttr::get(
-          ctx, idx, newRetType.getEncoding(), enumToType((type)));
-      auto newVType = RankedTensorType::get(
-          v.getType().getShape(), v.getType().getElementType(), newVEncoding);
-      v = rewriter.create<ConvertLayoutOp>(v.getLoc(), newVType, v);
-
-      // Bitcast
-      auto vTypeFp8 = RankedTensorType::get(
-          vType.getShape(), rewriter.getFloat8E4M3FNType(), newVEncoding);
-      v = cast<TypedValue<RankedTensorType>>(
-          rewriter.create<BitcastOp>(v.getLoc(), vTypeFp8, v).getResult());
-
-      // Convert to bf16
-      auto vTypeBf16 = RankedTensorType::get(
-          vType.getShape(), rewriter.getBF16Type(), newVEncoding);
-      return rewriter.create<FpToFpOp>(v.getLoc(), vTypeBf16, v);
+      if (type == F8F6F4Type::E2M1) {
+        // A bit too dynamically typed...
+        // perhaps return ints in both cases?
+
+        auto retEnc = dyn_cast<NvidiaMmaEncodingAttr>(newRetType.getEncoding());
+        auto newVEncoding = DotOperandEncodingAttr::get(
+            ctx, idx, newRetType.getEncoding(), /*kWidth=*/4);
+        auto newVType = RankedTensorType::get(
+            vType.getShape(), vType.getElementType(), newVEncoding);
+        return rewriter.create<ConvertLayoutOp>(v.getLoc(), newVType, v);
+      } else {
+        auto newVEncoding = DotOperandEncodingAttr::get(
```

Comment:
nit: assert that this is a fp8 type?

Reply:
Done, although it's a bit redundant, as we are already asserting this at the beginning of the function and in `semantics.py`.